backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -619,7 +635,7 @@ class SubmissionDatabaseService(
             }
         }
 
-        val filesToPublish = this.selectFilesForAccessionVersions(accessionVersionsToUpdate)
+        val filesToPublish = this.selectFilesToPublishForAccessionVersions(accessionVersionsToUpdate)
```

Comment:
sorry I only had a look at this now - what is the transaction level here?

Is the pipeline able to upload new prepro versions in parallel? could we have the case where prepro version 2 gets uploaded at the same time (or after) the user clicks approve? `selectFilesToPublishForAccessionVersions` doesn't see the new prepro version when it starts (this looks like just a read operation so it won't stop concurrent writes) and thus the second version will not be published

Reply:
thanks for pointing it out again!